PWGCF/Femto3D/TableProducer/singleTrackSelector.cxx

```diff
@@ -0,0 +1,180 @@
+                       ((applyEvSel.node() == 2) && (aod::evsel::sel8 == true));
+  Filter vertexFilter = ((o2::aod::collision::posZ < 15.f) && (o2::aod::collision::posZ > -15.f));
+  Filter trackFilter = ((o2::aod::track::itsChi2NCl <= 36.f) && (o2::aod::track::itsChi2NCl >= 0.f) && (o2::aod::track::tpcChi2NCl >= 0.f) && (o2::aod::track::tpcChi2NCl <= 4.f));
+
+  int mRunNumber = 0;
+  float d_bz = 0.f;
+
+  std::vector<int> particlesToKeep;
+  std::vector<int> particlesToReject;
+
+  void init(InitContext& context)
+  {
+
+    particlesToKeep = _particlesToKeep;
+    particlesToReject = _particlesToReject;
+
+    ccdb->setURL(ccdburl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    ccdb->setFatalWhenNull(false);
+  }
+
+  void initCCDB(aod::BCsWithTimestamps::iterator const& bc) // inspired by PWGLF/TableProducer/lambdakzerobuilder.cxx
+  {
+    if (mRunNumber == bc.runNumber()) {
+      return;
+    }
+    d_bz = 0.f;
+
+    auto run3grp_timestamp = bc.timestamp();
+    o2::parameters::GRPObject* grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>(grpPath, run3grp_timestamp);
+    o2::parameters::GRPMagField* grpmag = 0x0;
+    if (grpo) {
+      o2::base::Propagator::initFieldFromGRP(grpo);
+      // Fetch magnetic field from ccdb for current collision
+      d_bz = grpo->getNominalL3Field();
+      LOG(info) << "Retrieved GRP for timestamp " << run3grp_timestamp << " with magnetic field of " << d_bz << " kZG";
+    } else {
+      grpmag = ccdb->getForTimeStamp<o2::parameters::GRPMagField>(grpmagPath, run3grp_timestamp);
+      if (!grpmag) {
+        LOG(fatal) << "Got nullptr from CCDB for path " << grpmagPath << " of object GRPMagField and " << grpPath << " of object GRPObject for timestamp " << run3grp_timestamp;
+      }
+      o2::base::Propagator::initFieldFromGRP(grpmag);
+      // Fetch magnetic field from ccdb for current collision
+      d_bz = std::lround(5.f * grpmag->getL3Current() / 30000.f);
+      LOG(info) << "Retrieved GRP for timestamp " << run3grp_timestamp << " with magnetic field of " << d_bz << " kZG";
+    }
+    mRunNumber = bc.runNumber();
+    d_bz = 0.1 * d_bz;
+  }
+
+  void process(soa::Filtered<Coll>::iterator const& collision, soa::Filtered<Trks> const& tracks, aod::BCsWithTimestamps const&)
+  {
+
+    tableRow.reserve(tracks.size());
+
+    // auto collision = collisions.begin();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    initCCDB(bc);
+
+    tableRowColl(collision.multTPC(),
+                 collision.posZ(),
+                 d_bz);
+
+    for (auto& track : tracks) {
+
+      for (auto i : particlesToReject) {
+        // if satisfied, want to continue in the upper loop (over tracks) -- skip the current track
+        // cannot use simple 'continue' since it will be applied to the current loop, so have to use 'goto'
+        if (TOFselection(track, std::make_pair(i, rejectWithinNsigmaTOF))) {
+          goto skip;
```

Comment:
Please, no `goto`
Use meaningful name variables to control the execution flow

Reply:
I put a flag instead. No 'goto' anymore.